datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -823,11 +819,11 @@ macro_rules! get_data_page_statistics {
                     Float16Array::from_iter(
                         [<$stat_type_prefix Float16DataPageStatsIterator>]::new($iterator)
                             .map(|x| {
-                                x.into_iter().filter_map(|x| {
-                                    x.and_then(|x| Some(from_bytes_to_f16(x.data())))
+                                x.into_iter().map(|x| {
+                                    x.and_then(|x| from_bytes_to_f16(x.data()))
                                 })
                             })
-                            .flatten()
+                            .flatten().collect::<Vec<_>>()
```

Comment:
`collect` may be unnecessary here?

Reply:
That is an excellent call. It looks to me like this change adds `collect` to make this use consistent with the other arms. I tried removing `collect` locally and indeed the tests still pass

 I think `collect` is necessary in some cases to avoid a panic due to the iterator was not sized

like when I removed it from the BooleanArray I got an error like this

```
Iterator must be sized
thread 'parquet::arrow_statistics::test_data_page_stats_with_all_null_page' panicked at /Users/andrewlamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-52.1.0/src/array/boolean_array.rs:407:33:
Iterator must be sized
stack backtrace:
```

Reply:
Good catch - thanks.